File: 그리디/countingMoney.c
#include <stdio.h>
void greedyChange(int);
int main(void){
    int prize;
    int paidMoney;
    printf("물건의 가격을 입력하세요 : ");
    scanf("%d",&prize);
    printf("지불한 금액을 입력하세요 : ");
    scanf("%d",&paidMoney);
    greedyChange(paidMoney-prize);
    return 0;
}
void greedyChange(int money){
    if(money<0){
        printf("돈이 부족합니다.");
        return;
    }
    printf("\n총 거스름돈 -> %d원\n",money);
    int units[8]={50000,10000,5000,1000,500,100,50,10};
    int num[8]={0};
    for(int i=0;i<8;i++){
        num[i]=money/units[i];
        money-=units[i]*num[i];
    }
    for(int i=0;i<8;i++){
        printf("%5d원 %2d개\n",units[i],num[i]);
    }
}

File: 그리디/countingMoney.py
def greedyChange(money):
    if money<0:
        print("돈이 부족합니다.")
        return
    print("총 거스름돈 -> %d원"%(money))
    units=[50000,10000,5000,1000,500,100,50,10]
    num=[0,0,0,0,0,0,0,0]
    for i in range(len(units)):
        num[i]=int(money/units[i])
        money-=num[i]*units[i]
    for i in range(len(units)):
        print("%5d원 %2d개"%(units[i],num[i]))
prize=int(input("물건의 가격을 입력하세요 : "))
paidMoney=int(input("지불한 가격을 입력하세요 : "))
greedyChange(paidMoney-prize)